### PR TITLE
Check that s3 bucket has been created

### LINKED ISSUE #8395: NoSuchBucket on aws getting started
I ran into this when trying out k8s on AWS via kube-up:

```sh
Creating kubernetes-staging-4e8025693080b690b0bb0a08b3fc1d98
make_bucket: s3://kubernetes-staging-4e8025693080b690b0bb0a08b3fc1d98/
+++ Staging server tars to S3 Storage: kubernetes-staging-4e8025693080b690b0bb0a08b3fc1d98/devel
upload failed: server/kubernetes-server-linux-amd64.tar.gz to s3://kubernetes-staging-4e8025693080b690b0bb0a08b3fc1d98/devel/kubernetes-server-linux-amd64.tar.gz
A client error (NoSuchBucket) occurred when calling the UploadPart operation: The specified bucket does not exist
```

I suspect it's mainly that the newly created bucket takes some time to be usable. A retry worked fine.

Figured I'd at least open this to mention it and see if enough others are hitting it to make it worth building in a check loop for bucket creation.

### PR DESCRIPTION
**Pull request by matschaffer**
Fixes #8395

### CODE REVIEW

**Comment by matschaffer** [comment]
I signed it!


-Mat

about.me/matschaffer

File: cluster/aws/util.sh [hunk 1]
```diff
@@ -270,6 +270,23 @@ function upload-server-tars() {
     # We default to us-east-1 because that's the canonical region for S3,
     # and then the bucket is most-simply named (s3.amazonaws.com)
     aws s3 mb "s3://${AWS_S3_BUCKET}" --region ${AWS_S3_REGION}
+
+    local attempt=0
+    while true; do
+      if ! aws s3 ls "s3://${AWS_S3_BUCKET}" > /dev/null 2>&1; then
+        if (( attempt > 5 )); then
+          echo
+          echo -e "Unable to confirm bucket creation." >&2
```
Comment (justinsb) [inline cluster/aws/util.sh]: Nit: I don't think we need the -e here (or two below).  Unless you want to add ${color_red} ... ${color_norm}.  But it is harmless.

**Comment by justinsb** [comment]
LGTM - thanks @matschaffer

File: cluster/aws/util.sh (hunk 1 above)
Comment (matschaffer) [inline cluster/aws/util.sh]: Ah. Fair point. I was copying from the minion check. Which looks like it was copied from the salt-master check. Which actually uses `-e` for a reason.

**Pull request merged**